### PR TITLE
Fix most GCC 8 warnings

### PR DESCRIPTION
**Pull request by amadio**

### CODE REVIEW

**Comment by gganis** [comment]
@amadio : The fixes in TAuthenticate, TSecContext, rpdutils look good to me. Thanks!
Isn't there something to fix at THostAuth:200? See <https://epsft-jenkins.cern.ch/view/ROOT/job/root-nightly-master/LABEL=ROOT-fedora29,SPEC=cxx14/1977/parsed_console/>

File: core/base/src/TSystem.cxx [hunk 1]
```diff
@@ -1146,7 +1146,7 @@ Bool_t TSystem::ExpandFileName(const char *fname, char *xname, const int kBufSiz
    } else if (c[0] == '~' && c[1] != '/') { // ~user case
       n = strcspn(c+1, "/ ");
       buff[0] = 0;
-      strncat(buff, c+1, n);
+      (void) strlcat(buff, std::string(c+1).c_str(), sizeof(buff) - 1);
```
Comment (pcanal) [inline core/base/src/TSystem.cxx]: This change the semantic of the code.  It needs to stop copying the string at 'n'.  so more like
```
strlcat(buf, c+1, std::min(n, sizeof buff - 1);
```

Comment (amadio) [reply core/base/src/TSystem.cxx]: Yes, I remember that... I thought you had fixed that one. I will change.

**Comment by pcanal** [comment]
Aren't the warning in ROOT-fedora27/noimt real/new?

**Comment by amadio** [comment]
Yeah, sorry, I will fix when I get back to work today. It's a warning about unknown ignored warnings... So annoying.

**Pull request merged**